include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,321 @@
+   // The timestamps are in seconds since UNIX EPOCH, see C's time_t time(time_t *).
+   // If one of the time isn't known, then set it to 0.
+   // If this function is not called, then the indexer may look at the file's creation and
+   // modification time.
+   void(CLAP_ABI *set_timestamps)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  uint64_t creation_time,
+                                  uint64_t modification_time);
+
+   // Adds a feature to the preset.
+   //
+   // The feature string is arbitrary, it is the indexer's job to understand it and remap it to its
+   // internal categorization and tagging system.
+   //
+   // Examples:
+   // kick, drum, tom, snare, clap, cymbal, bass, lead, metalic, hardsync, crossmod, acid,
+   // distorted, drone, pad, dirty, etc...
+   void(CLAP_ABI *add_feature)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *feature);
+
+   // Adds extra information to the metadata.
+   void(CLAP_ABI *add_extra_info)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  const char                                           *key,
+                                  const char                                           *value);
+} clap_preset_discovery_metadata_receiver_t;
+
+typedef struct clap_preset_discovery_filetype {
+   const char *name;
+   const char *description;
+
+   // `.' isn't included in the string.
+   // If empty or NULL then every file should be matched.
+   const char *file_extension;
+} clap_preset_discovery_filetype_t;
+
+// Defines a place in which to search for presets
+typedef struct clap_preset_discovery_location {
+   uint32_t    flags; // see enum clap_preset_discovery_flags
+   const char *name;  // name of this location
+
+   // URI
+   // - file:/// for pointing to a file or directory; directories are scanned recursively
+   //   eg: file:///home/abique/.u-he/Diva/presets/ (on Linux)
+   //   eg: file:///C:/Users/abique/Documents/u-he/Diva.data/presets/ (on Windows)
+   // - plugin://<clap-plugin-id> for presets which are bundled inside the plugin itself
+   const char *uri;
+} clap_preset_discovery_location_t;
+
+// Describes an installed sound pack.
+typedef struct clap_preset_discovery_soundpack {
+   uint64_t    flags;        // see enum clap_preset_discovery_flags
+   const char *id;           // sound pack identifier
+   const char *name;         // name of this sound pack
+   const char *description;  // reasonably short description of the sound pack
+   const char *homepage_url; // url to the pack's homepage
+   const char *vendor;       // sound pack's vendor
+   const char *image_uri;    // may be an image on disk or from an http server
+
+   // release date, in numbe of seconds since UNIX EPOCH, 0 if unavailable
+   uint64_t release_timestamp;
```

Comment:
This made me think whether it might be helpful to have a version number for soundpacks (?) Probably useful for soundpack updates.

Reply:
What would the host do with that?
Release timestamp could be used in a search query like "All soundpacks released in 2003".
But would the user search for "all soundpacks at fourth update level"? I doubt it.